### PR TITLE
Added community service module

### LINKED ISSUE #129: Collegist's community service module

### PR DESCRIPTION
**Pull request by sszajbely**
Users are able to track their community services in the Collegium. Features:
- Everyone with the collegist role can view the module.
- As default everyone sees the records where they are the requester or approver. This view is devided into semesters.
- Everyone with the current semester active can create a new request.
- Approvers can be chosen from the leaders of the Student's Council and the leaders of the committees.
- The chosen approver receives an email. If they approve the requester receives an email.
- The approvers (defined earlier) can search for every active collegist's past community services.

### CODE REVIEW

File: app/Policies/CommunityServicePolicy.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Determine whether the user is the approver for the given community service.
+     * @param User $user
+     * @param CommunityService $communityService
+     * @return bool
+     */
+    public function approve(User $user, CommunityService $communityService)
+    {
+        if ($communityService->approved || !$communityService->semester->isCurrent()) {
```
Comment (kdmnk) [inline app/Policies/CommunityServicePolicy.php]: You can check for the 2 week interval here.

File: resources/views/emails/community_service_requested.blade.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+@component('mail::message')
+    <h1>Kedves {{ $recipient->name }}!</h1>
+    <p>
+        {{ $requester->name }} új közösségi tevékenységet hozott létre a következő leírással: <br>
+        "{{ $description }}"<br>
+        Kérlek fogadd el vagy utasítsd el a tevékenységet a következő linken a táblázatban: <br>
+        @component('mail::button', ['url'=> route('community_service')])
```
Comment (kdmnk) [inline resources/views/emails/community_service_requested.blade.php:7]: You can forward this to the search route with the chosen user

File: resources/views/student-council/community-service/search.blade.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                                    <td>{{ $communityService->created_at->format('Y. m. d.') }}</td>
+                                    <td>{{ $communityService->requester->name }}</td>
+                                    <td>{{ $communityService->approver->name }}</td>
+                                    <td>{{ $communityService->status }}</td>
+                                </tr>
+                            @endforeach
+                        </tbody>
+                    </table>
+                </div>
+            </div>
```
Comment (kdmnk) [inline resources/views/student-council/community-service/search.blade.php]: Create a new file for this table, and include that in both used places.

**Comment by kdmnk** [comment]
I made some refactorings and introduced a new state for pending status because I found it confusing without it. 

Make sure to add the missing locales, I forget that.
Also, you can even merge the two pages into one, if you want. You just need to pass "$request->searched_user ?? $request->user()" for the query in the controller (and modify the policy). 

You might want a function also for rejecting the requests.

**Review by sszajbely — COMMENTED** [review]
I intentionally left out the status on the search site, but it's not a big deal.

File: resources/views/student-council/community-service/search.blade.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
+@extends('layouts.app')
+
+@section('title')
+<a href="#!" class="breadcrumb">@lang('role.student-council')</a>
+<a href="{{route('community_service')}}" style="cursor: pointer" class="breadcrumb">@lang('community-service.community-service')</a>
+<a href="#!" class="breadcrumb">@lang('community-service.community-service')</a>
```
Comment (kdmnk) [inline resources/views/student-council/community-service/search.blade.php]: Display "search" or the searched user name

File: resources/views/student-council/community-service/search.blade.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        <form action={{ route('community_service.search')}} method="GET">
+            <div class="row center">
+                <x-input.select m=12 l=12 id="requester" :elements="\App\Models\User::active()->get()" :text="__('community-service.requester')"/>
+                <x-input.button text="general.search"/>
+            </div>
+        </form>
+    </div>
+</div>
+@foreach($semesters as $semester)
+<div class="col s12">
```
Comment (kdmnk) [inline resources/views/student-council/community-service/search.blade.php]: delete the col div

File: app/Policies/CommunityServicePolicy.php (hunk 1 above)
Comment (sszajbely) [reply app/Policies/CommunityServicePolicy.php]: I don't want to, to be honest. Just want to scare the approvers.

File: resources/views/emails/community_service_requested.blade.php (hunk 2 above)
Comment (sszajbely) [reply resources/views/emails/community_service_requested.blade.php:7]: I don't know how I can do that.

**Review by kdmnk — APPROVED** [review]

**Pull request merged**